cps/tasks/metadata_extract.py

```diff
@@ -110,7 +110,7 @@ def _update_metadata(self, requested_urls):
         requested_urls = {url: requested_urls[url] for url in requested_urls.keys() if "shorts" not in url and url not in failed_urls}
 
     def _calculate_views_per_day(self, requested_urls, conn):
-        now = datetime.now()
+        now = datetime.now(timezone.utc)
         for requested_url in requested_urls.keys():
             try:
                 view_count = conn.execute("SELECT view_count FROM media WHERE path = ?", (requested_url,)).fetchone()[0]
```

Comment:
On line 117, what's the type of the time_uploaded field in the database?
Could you point me at the code writing this field, to ensure it's actually a POSIX timestamp (aka UNIX epoch)?

Also, datestamp.utcfromtimestamp() creates a naive datestamp, so if this is really a UNIX epoch, we should write:

```python
time_uploaded_raw = conn.execute("...").fetchone()[0]
time_uploaded = datetime.fromtimestamp(time_uploaded_raw, timezone.utc)
```

Reply:
I found the code writing `time_uploaded` at line 186:

```python
conn.execute("UPDATE playlists SET path = ? WHERE path = ?", (f"{self.media_url}&timestamp={int(datetime.now().timestamp())}", self.media_url))
```

Since `int(datetime.now().timestamp())` is the same of `int(time.time())`, this can be simplified.

Note that most SQL databases have types like TIMESTAMP and DATETIME, but SQLite by convention uses integer POSIX timestamps. So we're already storing the correct number here, but we could directly use `int(time.time())` instead of converting the system time to datetime and then back to seconds.

Reply:
I agree here, but will this be consistent across the code base which uses datetime objects to render tasks status. Some tasks are triggered by cron job and some are run upon uploading. For reporting purpose, these datetime objects are used in th Start Time column of the tasks page.

![image](https://github.com/iiab/calibre-web/assets/16546989/d75fce48-512d-4558-9da5-e3b0c7a86c64)